packages/component-library/components/Button/Button.elm

```diff
@@ -145,19 +169,26 @@ onClickAttribs (Config config) =
 
                         Nothing ->
                             True
+
+                decoder =
+                    Json.map
+                        (\m ->
+                            { message = m
+                            , stopPropagation = False
+                            , preventDefault = preventDefault
+                            }
+                        )
+                        (Json.succeed msg)
             in
-            [ onWithOptions
-                "click"
-                { defaultOptions | preventDefault = preventDefault }
-                (Json.succeed msg)
+            [ HtmlEvents.custom "click" decoder
```

Comment:
I really appreciate this change and removing the 0.18 compatibility dependency - thanks! Could I be a pain and ask you to split this into a separate PR though? It's unrelated to setFocus and setBlur. Just in case this change causes issues, we need to be able to roll it back easily (at the very least it should be a separate commit).

It also makes it easier to categorise the change. adding `setBlur` and `setFocus` is a `feat`, but changing the implementation of `onClick` is a `fix`. It makes it easier to release changes as it's clearer what's going to change or not.

Reply:
👍